include/effectmanager.h

```diff
@@ -72,25 +73,32 @@ class EffectManager : public IJSONSerializable
     std::vector<LEDStripEffect*> _vEffects;
     size_t _cEnabled = 0;
 
-    size_t _iCurrentEffect;
+    size_t _iCurrentEffect = 0;
     uint _effectStartTime;
-    uint _effectInterval;
+    uint _effectInterval = 0;
     bool _bPlayAll;
     bool _bShowVU = true;
     CRGB lastManualColor = CRGB::Red;
+    bool _clearRemoteEffectWhenExpired = false;
 
     std::unique_ptr<bool[]> _abEffectEnabled;
     std::shared_ptr<GFXTYPE> * _gfx;
-    std::shared_ptr<LEDStripEffect> _ptrRemoteEffect = nullptr;
+    std::shared_ptr<LEDStripEffect> _ptrRemoteEffect;
 
-    void construct() 
+    void construct(bool clearRemoteEffect)
     {
         _bPlayAll = false;
-        _iCurrentEffect = 0;
-        _effectStartTime = millis();
+
+        if (clearRemoteEffect && _ptrRemoteEffect)
+        {
+            _clearRemoteEffectWhenExpired = true;
+
+            // This is a hacky way to ensure that we start the correct effect after the temporary one
+            _iCurrentEffect--;
+        }
     }
 
-    void ClearEffects() 
+    void ClearEffects()
     {
         for (auto effect : _vEffects)

```

Comment:
Watch for a merge conflict here, we no longer need to delete the items individually now that they're in vectors...

Reply:
They have to still be deleted here, because in this version the contents of the vector are still raw pointers. Beyond that you're right, there will be some merge conflicts to address when we merge the safe pointer switch.